Use dedup-id instead of raw-object

cammellos opened the pull request:
This is a backward/forward compatible change with status-go.
We are changing the way messages are confirmed from passing the
raw-object to status-go to a dedup-id instead, which needs to be sent
back.
Based on the response from statu-go we detect whether they are ids or
object and act accordingly.

status: ready

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://docs.status.im/docs/build_status.html))?

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (29)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | 1cfe47fc | [#1](https://ci.status.im/job/status-react/job/prs/job/ios-e2e/job/PR-7475/1/) | 2019-02-12 08:50:03 | ~7 min | `ios-e2e` | [:page_facing_up: log](https://ci.status.im/job/status-react/job/prs/job/ios-e2e/job/PR-7475/1/consoleText) |
| :x: | 1cfe47fc | [#1](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-7475/1/) | 2019-02-12 08:51:21 | ~8 min | `ios` | [:page_facing_up: log](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-7475/1/consoleText) |
| :x: | 1cfe47fc | [#1](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-7475/1/) | 2019-02-12 08:56:25 | ~13 min | `android-e2e` | [:page_facing_up: log](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-7475/1/consoleText) |
| :x: | 1cfe47fc | [#1](https://ci.status.im/job/status-react/job/prs/job/android/job/PR-7475/1/) | 2019-02-12 08:57:14 | ~14 min | `android` | [:page_facing_up: log](https://ci.status.im/job/status-react/job/prs/job/android/job/PR-7475/1/consoleText) |
| :heavy_check_mark: | 1cfe47fc | [#1](https://ci.status.im/job/status-react/job/prs/job/macos/job/PR-7475/1/) | 2019-02-12 08:58:09 | ~15 min | `macos` | [:package: dmg](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190212-084236-1cfe47-pr.dmg) |
| :heavy_check_mark: | 1cfe47fc | [#1](https://ci.status.im/job/status-react/job/prs/job/windows/job/PR-7475/1/) | 2019-02-12 09:04:14 | ~21 min | `windows` | [:package: exe](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190212-084237-1cfe47-pr.exe) |
| :heavy_check_mark: | 1cfe47fc | [#1](https://ci.status.im/job/status-react/job/prs/job/linux/job/PR-7475/1/) | 2019-02-12 09:09:48 | ~27 min | `linux` | [:package: App](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190212-084235-1cfe47-pr.AppImage) |
| :x: | 1cfe47fc | [#2](https://ci.status.im/job/status-react/job/prs/job/android/job/PR-7475/2/) | 2019-02-12 10:13:08 | ~21 sec | `android` | [:page_facing_up: log](https://ci.status.im/job/status-react/job/prs/job/android/job/PR-7475/2/consoleText) |
| | | | | | | |
| :heavy_check_mark: | fc9af586 | [#2](https://ci.status.im/job/status-react/job/prs/job/ios-e2e/job/PR-7475/2/) | 2019-02-12 10:31:38 | ~15 min | `ios-e2e` | [:package: pkg](https://status-im.ams3.digitaloceanspaces.com/StatusIm-190212-101634-fc9af5-e2e.app.zip) |
| :heavy_check_mark: | fc9af586 | [#2](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-7475/2/) | 2019-02-12 10:33:26 | ~16 min | `android-e2e` | [:package: apk](https://status-im.ams3.digitaloceanspaces.com/StatusIm-190212-101632-fc9af5-e2e.apk) |
| :heavy_check_mark: | fc9af586 | [#2](https://ci.status.im/job/status-react/job/prs/job/windows/job/PR-7475/2/) | 2019-02-12 10:34:27 | ~17 min | `windows` | [:package: exe](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190212-101638-fc9af5-pr.exe) |
| :heavy_check_mark: | fc9af586 | [#2](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-7475/2/) | 2019-02-12 10:36:10 | ~19 min | `ios` | [:package: ipa](https://i.diawi.com/MBo1A9) |
| :heavy_check_mark: | fc9af586 | [#2](https://ci.status.im/job/status-react/job/prs/job/linux/job/PR-7475/2/) | 2019-02-12 10:36:44 | ~20 min | `linux` | [:package: App](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190212-101636-fc9af5-pr.AppImage) |
| :heavy_check_mark: | fc9af586 | [#2](https://ci.status.im/job/status-react/job/prs/job/macos/job/PR-7475/2/) | 2019-02-12 10:36:46 | ~20 min | `macos` | [:package: dmg](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190212-101637-fc9af5-pr.dmg) |
| :heavy_check_mark: | fc9af586 | [#3](https://ci.status.im/job/status-react/job/prs/job/android/job/PR-7475/3/) | 2019-02-12 10:37:42 | ~21 min | `android` | [:package: apk](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190212-101633-fc9af5-pr.apk) |
| | | | | | | |
| :heavy_check_mark: | 27e4e9f6 | [#3](https://ci.status.im/job/status-react/job/prs/job/ios-e2e/job/PR-7475/3/) | 2019-02-15 08:13:16 | ~15 min | `ios-e2e` | [:package: pkg](https://status-im.ams3.digitaloceanspaces.com/StatusIm-190215-075755-27e4e9-e2e.app.zip) |
| :heavy_check_mark: | 27e4e9f6 | [#3](https://ci.status.im/job/status-react/job/prs/job/macos/job/PR-7475/3/) | 2019-02-15 08:14:00 | ~16 min | `macos` | [:package: dmg](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190215-075758-27e4e9-pr.dmg) |
| :heavy_check_mark: | 27e4e9f6 | [#3](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-7475/3/) | 2019-02-15 08:14:55 | ~16 min | `ios` | [:package: ipa](https://i.diawi.com/89kvhM) |
| :heavy_check_mark: | 27e4e9f6 | [#3](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-7475/3/) | 2019-02-15 08:15:37 | ~17 min | `android-e2e` | [:package: apk](https://status-im.ams3.digitaloceanspaces.com/StatusIm-190215-075753-27e4e9-e2e.apk) |
| :heavy_check_mark: | 27e4e9f6 | [#4](https://ci.status.im/job/status-react/job/prs/job/android/job/PR-7475/4/) | 2019-02-15 08:16:30 | ~18 min | `android` | [:package: apk](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190215-075754-27e4e9-pr.apk) |
| :heavy_check_mark: | 27e4e9f6 | [#3](https://ci.status.im/job/status-react/job/prs/job/windows/job/PR-7475/3/) | 2019-02-15 08:20:29 | ~22 min | `windows` | [:package: exe](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190215-075758-27e4e9-pr.exe) |
| :heavy_check_mark: | 27e4e9f6 | [#3](https://ci.status.im/job/status-react/job/prs/job/linux/job/PR-7475/3/) | 2019-02-15 08:29:07 | ~31 min | `linux` | [:package: App](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190215-075757-27e4e9-pr.AppImage) |
| | | | | | | |
| :heavy_check_mark: | 14fe2a1a | [#4](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-7475/4/) | 2019-02-15 11:26:46 | ~15 min | `ios` | [:package: ipa](https://i.diawi.com/WqZe7y) |
| :heavy_check_mark: | 14fe2a1a | [#4](https://ci.status.im/job/status-react/job/prs/job/macos/job/PR-7475/4/) | 2019-02-15 11:26:55 | ~15 min | `macos` | [:package: dmg](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190215-111138-14fe2a-pr.dmg) |
| :heavy_check_mark: | 14fe2a1a | [#4](https://ci.status.im/job/status-react/job/prs/job/ios-e2e/job/PR-7475/4/) | 2019-02-15 11:27:07 | ~15 min | `ios-e2e` | [:package: pkg](https://status-im.ams3.digitaloceanspaces.com/StatusIm-190215-111136-14fe2a-e2e.app.zip) |
| :heavy_check_mark: | 14fe2a1a | [#5](https://ci.status.im/job/status-react/job/prs/job/android/job/PR-7475/5/) | 2019-02-15 11:27:55 | ~16 min | `android` | [:package: apk](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190215-111135-14fe2a-pr.apk) |
| :heavy_check_mark: | 14fe2a1a | [#4](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-7475/4/) | 2019-02-15 11:28:42 | ~17 min | `android-e2e` | [:package: apk](https://status-im.ams3.digitaloceanspaces.com/StatusIm-190215-111133-14fe2a-e2e.apk) |
| :heavy_check_mark: | 14fe2a1a | [#4](https://ci.status.im/job/status-react/job/prs/job/windows/job/PR-7475/4/) | 2019-02-15 11:32:58 | ~21 min | `windows` | [:package: exe](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190215-111139-14fe2a-pr.exe) |
| :heavy_check_mark: | 14fe2a1a | [#4](https://ci.status.im/job/status-react/job/prs/job/linux/job/PR-7475/4/) | 2019-02-15 11:40:31 | ~28 min | `linux` | [:package: App](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190215-111137-14fe2a-pr.AppImage) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 51886314 | [#5](https://ci.status.im/job/status-react/job/prs/job/macos/job/PR-7475/5/) | 2019-02-15 15:19:31 | ~13 min | `macos` | [:package: dmg](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190215-150533-518863-pr.dmg) |
| :heavy_check_mark: | 51886314 | [#5](https://ci.status.im/job/status-react/job/prs/job/ios-e2e/job/PR-7475/5/) | 2019-02-15 15:20:43 | ~15 min | `ios-e2e` | [:package: pkg](https://status-im.ams3.digitaloceanspaces.com/StatusIm-190215-150528-518863-e2e.app.zip) |
| :heavy_check_mark: | 51886314 | [#5](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-7475/5/) | 2019-02-15 15:21:40 | ~16 min | `ios` | [:package: ipa](https://i.diawi.com/NSGc1e) |
| :heavy_check_mark: | 51886314 | [#6](https://ci.status.im/job/status-react/job/prs/job/android/job/PR-7475/6/) | 2019-02-15 15:24:26 | ~18 min | `android` | [:package: apk](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190215-150527-518863-pr.apk) |
| :heavy_check_mark: | 51886314 | [#5](https://ci.status.im/job/status-react/job/prs/job/windows/job/PR-7475/5/) | 2019-02-15 15:24:36 | ~19 min | `windows` | [:package: exe](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190215-150533-518863-pr.exe) |
| :heavy_check_mark: | 51886314 | [#5](https://ci.status.im/job/status-react/job/prs/job/linux/job/PR-7475/5/) | 2019-02-15 15:25:24 | ~19 min | `linux` | [:package: App](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190215-150530-518863-pr.AppImage) |
| :heavy_check_mark: | 51886314 | [#5](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-7475/5/) | 2019-02-15 15:29:35 | ~24 min | `android-e2e` | [:package: apk](https://status-im.ams3.digitaloceanspaces.com/StatusIm-190215-150526-518863-e2e.apk) |
| | | | | | | |
| :heavy_check_mark: | 574d19e7 | [#6](https://ci.status.im/job/status-react/job/prs/job/ios-e2e/job/PR-7475/6/) | 2019-02-17 18:42:49 | ~14 min | `ios-e2e` | [:package: pkg](https://status-im.ams3.digitaloceanspaces.com/StatusIm-190217-182801-574d19-e2e.app.zip) |
| :heavy_check_mark: | 574d19e7 | [#7](https://ci.status.im/job/status-react/job/prs/job/android/job/PR-7475/7/) | 2019-02-17 18:43:52 | ~15 min | `android` | [:package: apk](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190217-182800-574d19-pr.apk) |
| :heavy_check_mark: | 574d19e7 | [#6](https://ci.status.im/job/status-react/job/prs/job/macos/job/PR-7475/6/) | 2019-02-17 18:43:58 | ~15 min | `macos` | [:package: dmg](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190217-182803-574d19-pr.dmg) |
| :heavy_check_mark: | 574d19e7 | [#6](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-7475/6/) | 2019-02-17 18:44:42 | ~16 min | `ios` | [:package: ipa](https://i.diawi.com/NVevZL) |
| :heavy_check_mark: | 574d19e7 | [#6](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-7475/6/) | 2019-02-17 18:44:44 | ~16 min | `android-e2e` | [:package: apk](https://status-im.ams3.digitaloceanspaces.com/StatusIm-190217-182759-574d19-e2e.apk) |
| :heavy_check_mark: | 574d19e7 | [#6](https://ci.status.im/job/status-react/job/prs/job/windows/job/PR-7475/6/) | 2019-02-17 18:47:02 | ~18 min | `windows` | [:package: exe](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190217-182804-574d19-pr.exe) |
| :heavy_check_mark: | 574d19e7 | [#6](https://ci.status.im/job/status-react/job/prs/job/linux/job/PR-7475/6/) | 2019-02-17 18:48:05 | ~20 min | `linux` | [:package: App](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190217-182803-574d19-pr.AppImage) |

ghost commented:
`mobile_files/yarn.lock` changed. Pinging @jakubgs, @mandrigin, and @corpetty

ghost commented:
`mobile_files/yarn.lock` changed. Pinging @jakubgs, @mandrigin, and @corpetty

ghost commented:
`mobile_files/yarn.lock` changed. Pinging @jakubgs, @mandrigin, and @corpetty

yenda commented on src/status_im/transport/message/core.cljs:
```diff
@@ -12,10 +13,24 @@
   (let [blocked-contacts (get-in cofx [:db :contacts/blocked] #{})
-        {:keys [payload sig timestamp ttl]} (js->clj js-message :keywordize-keys true)
+        {{:keys [payload sig timestamp ttl]} :message
+         dedup-id :id
```
Could you add a docstring to explain in the code what the dedup-id is for and where it is coming from, not sure where is the best place to do that maybe here.

cammellos replied:
sure, good idea

rasom commented:
typo `statu-go` in commit message

ghost commented:
`mobile_files/yarn.lock` changed. Pinging @jakubgs, @mandrigin, and @corpetty

ghost commented:
`mobile_files/yarn.lock` changed. Pinging @jakubgs, @mandrigin, and @corpetty

cammellos committed:
Use dedup-id instead of raw-payload

This is a backward/forward compatible change with status-go.
We are changing the way messages are confirmed from passing the
raw-object to status-go to a dedup-id instead, which needs to be sent
back.
Based on the response from status-go we detect whether they are ids or
object and act accordingly.

Signed-off-by: Andrea Maria Piana <andrea.maria.piana@gmail.com>

ghost commented:
`mobile_files/yarn.lock` changed. Pinging @jakubgs, @mandrigin, and @corpetty

ghost commented:
`mobile_files/yarn.lock` changed. Pinging @jakubgs, @mandrigin, and @corpetty
